t/05-scheduler-dependencies.t

```diff
@@ -872,4 +874,22 @@ subtest 'slepos test workers' => sub {
     ok(_job_cloned_and_related($jobBS,  $jobT),   "jobBS and jobT");
 };
 
+# This enforces order in the processing of the nodes, to test PR#1623
+my $ordered_sort = sub {
+    my ($self, $result_set, $condition, $attrs) = @_;
+    $attrs->{order_by} = {-desc => 'id'};
+    return $self->$result_set->search($condition, $attrs);
+};
+
+my $unordered_sort = *OpenQA::Schema::Result::Jobs::search_for;
+my $tb             = Test::More->builder;
+my %tests          = ('duplicate' => $duplicate_test, 'slepos test workers' => $slepos_test_workers);
+while (my ($k, $v) = each %tests) {
+    no warnings 'redefine';
+    *OpenQA::Schema::Result::Jobs::search_for = $unordered_sort;
+    $tb->subtest("$k unordered", $v);
+    *OpenQA::Schema::Result::Jobs::search_for = $ordered_sort;
```

Comment:
In this way you are replacing the function, so the test will always excute your custom defined one, i suggest to wrap the original function, and applying the order after calling it so it gets actually hit.

Reply:
Done.